opentelemetry-sdk/src/trace/tracer.rs

```diff
@@ -172,19 +188,20 @@ impl opentelemetry::trace::Tracer for Tracer {
 
         // In order to accommodate use cases like `tracing-opentelemetry` we there is the ability
         // to use pre-sampling. Otherwise, the standard method of sampling is followed.
-        let sampling_decision = if let Some(sampling_result) = builder.sampling_result.take() {
-            self.process_sampling_result(sampling_result, parent_cx)
-        } else {
-            self.make_sampling_decision(
-                parent_cx,
-                trace_id,
-                &builder.name,
-                &span_kind,
-                builder.attributes.as_ref().unwrap_or(&Vec::new()),
-                builder.links.as_deref().unwrap_or(&[]),
-                provider.config(),
-            )
-        };
+        let (decision, flags, extra_attrs, trace_state) =
+            if let Some(sampling_result) = builder.sampling_result.take() {
+                self.process_sampling_result(sampling_result, parent_cx)
+            } else {
+                self.make_sampling_decision(
+                    parent_cx,
+                    trace_id,
+                    &builder.name,
+                    &span_kind,
+                    builder.attributes.as_ref().unwrap_or(&Vec::new()),
+                    builder.links.as_deref().unwrap_or(&[]),
+                    provider.config(),
+                )
+            };
```

Comment:
I don't see we use `trace_state` in this part. Maybe we can just extract it before we process sampling result or make a sampling decision?